### PR TITLE
Multiple Uploads: Fix bug where I can't send regular files to attribute writter

### PR DESCRIPTION
**Pull request by nicooga**
In reference to https://github.com/lebedev-yury/carrierwave-base64/pull/20.

This fixes this:

~~~ruby
class Post
    mount_base64_uploaders :pictures, PictureUploader
end

Post.new pictures: [File.open(...)]
=> TypeError: no implicit conversion of nil into String
~~~
... so you can mix regular files and base64 strings. Also I added some tests, although the depend on installing carrierwave from master branch at github.

### CODE REVIEW

**Comment by lebedev-yury** [comment]
this looks already much better.

Could you please also remove ruby 1.9.3 from the `travis.yml`? Seems that the new carrierwave will drop support for it.

File: lib/carrierwave/base64/adapter.rb [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
+
+      private
+
+      def handle_base64_image_data(data)
+        return data unless data.present?
+
+        if data.is_a?(String) && data.strip.start_with?("data")
+          Carrierwave::Base64::Base64StringIO.new(data.strip)
+        elsif data.is_a?(Hash)
+          binding.pry
```
Comment (txssseal) [inline lib/carrierwave/base64/adapter.rb]: :-1:

**Comment by nicooga** [comment]
I learned the hard way that multiple file uploads are not designed to be
usable through a REST api.
The problem is there is no easy way to add or remove a file to the
attribute but by completely replacing the whole array.
My principal interest in this feature was that I could use it to send files
in base64 in a normal PUT request along side other model attributes, but
this bug (?) kills the purpose.

For the time being I just resorted to use a has_many with a model that has
a single mounted uploader.

**Comment by lebedev-yury** [comment]
@nicooga hi, sorry for just closing this PR, it was not active for some time.

About multiple files upload through an API, I understand your concern. It's probably not the best idea to upload several files in one batch to the API.

But the support for multifile uploads will still be added to the gem, just because carrierwave supports it.

**Pull request closed without merging**